packages/framework/tests/Unit/Model/Category/CategoryServiceTest.php

```diff
@@ -15,7 +16,7 @@ public function testCreateSetRoot()
         $categoryData = new CategoryData();
         $rootCategory = new Category($categoryData);
 
-        $categoryService = new CategoryService(new CategoryFactory());
+        $categoryService = new CategoryService(new CategoryFactory(new EntityNameResolver([])));
```

Comment:
This made me think whether we should make the argument in the constructor optional like this:

```php
    public function __construct(EntityNameResolver $entityNameResolver = null)
    {
        $this->entityNameResolver = $entityNameResolver ?? new EntityNameResolver([]);
    }
```
But I'm not sure if it's a thing we'd wanna do...

Reply:
That's common to see in Symfony code for historical, because they didn't use dependency injection for a many years.

But when this happens:

```diff
 class EntityNameResolver
 {
+    public function __construct(SomeNewDependency $someNewDependency)
+    {
+    }
 }
```

[the constructor hell](https://phpfashion.com/di-a-predavani-zavislosti#toc-constructor-hell) beings :fire: 

<br>

I don't have a context here, but what good is to create services manually and edit for every new dependency? Using DI container would prevent such manuall statics.

```diff
-$categoryService = new CategoryService(new CategoryFactory(new EntityNameResolver([])));
+$categoryService = $this->container->get(CategoryService::class);
```

Reply:
I know, I suggest it should be